app/controllers/course/survey/surveys_controller.rb

```diff
@@ -84,7 +84,8 @@ def survey_params
       :title, :description, :base_exp, :time_bonus_exp, :start_at, :bonus_end_at, :end_at,
       :published, :allow_response_after_end, :allow_modify_after_submit
     ]
-    fields << :anonymous if !@survey.anonymous || !@survey.has_student_response?
+    fields << :anonymous if action_name == 'create' || !@survey.anonymous ||
```

Comment:
@kxmbrian sorry to dig this up again, but would it be better to move this logic into the model instead? Can use a validation callback to prevent the changing of anonymous mode for surveys based on certain conditions.